core/src/main/java/hudson/model/Queue.java

```diff
@@ -433,6 +441,22 @@ public synchronized void load() {
         }
     }
 
+    private long getMaxNextBuildNumber() {
+        long maxNextBuildNumber = 0;
+        Jenkins jenkins = Jenkins.getInstance();
+
+        if (jenkins != null) {
+            for (String jobName : jenkins.getJobNames()) {
```

Comment:
surly just Jenkins.getAllItems(Job.class) ?

How does this interact with Job loading at startup?